google/cloud/storage/oauth2/google_credentials.cc

```diff
@@ -174,13 +174,15 @@ CreateAuthorizedUserCredentialsFromJsonContents(std::string const& contents) {
 StatusOr<std::shared_ptr<Credentials>>
 CreateServiceAccountCredentialsFromJsonFilePath(std::string const& path) {
   return CreateServiceAccountCredentialsFromJsonFilePath(
-      path, {}, google::cloud::optional<std::string>());
+      path, google::cloud::optional<std::set<std::string>>(),
+      google::cloud::optional<std::string>());
```

Comment:
You should rarely need to explicitly specify the "optional" type in calling code. So I think you should be able to write this more simply as:

```cc
return CreateServiceAccountCredentialsFromJsonFilePath(
    path, std::set<std::string>{}, std::string{});
```

And elsewhere.

Actually, I'm curious why this is passing the empty set and empty string as arguments instead of the `nullopt` argument? Or, it may be that this function is defined to call the inner one with the right defaults, meaning that it might be at *this* point that you set scopes *by default* to `GoogleOAuthScopeCloudPlatform()`.

Thoughts?

Reply:
The compiler complains that it can't correctly infer the types when I leave the optional wrapper off.  Also, a populated optional with an empty set (meaning a set of 0 scopes) is different than an empty optional (no scopes given, just use the default scope(s)).  And I figured it made sense to follow suit and use the same semantics for the string argument.

Reply:
Ahh, ok. Sorry. I think that *should* work, but I'll take a look at that separately and not hold up your PR anymore.